### PR TITLE
[OPIK-203] Filter our spans.usage with null values

### PR DESCRIPTION
**Pull request by BorisTkachenko**
## Details
Null value in payload leads to 500 error

Resolves #https://comet-ml.atlassian.net/browse/OPIK-203

## Testing
Tested with running Opik locally. Reproduced without the fix, and then checked that fix resolves this issue.
Added integration test.

### CODE REVIEW

File: apps/opik-backend/src/main/resources/liquibase/db-app-analytics/migrations/000007_change_spans_usage_value_to_nullable.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+--liquibase formatted sql
+--changeset thiagohora:change_decimal_precision
```
Comment (thiagohora) [inline apps/opik-backend/src/main/resources/liquibase/db-app-analytics/migrations/000007_change_spans_usage_value_to_nullable.sql]: wrong name

File: apps/opik-backend/src/main/resources/liquibase/db-app-analytics/migrations/000007_change_spans_usage_value_to_nullable.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+ALTER TABLE ${ANALYTICS_DB_DATABASE_NAME}.spans
+    MODIFY COLUMN usage Map(String, Nullable(Int32));
```
Comment (thiagohora) [inline apps/opik-backend/src/main/resources/liquibase/db-app-analytics/migrations/000007_change_spans_usage_value_to_nullable.sql]: We try as much as possible to avoid nulls because they have a performance hit in ClickHouse. @andrescrz, what would be better here? Using null, or making it as a string and using empty as null and enforcing number at the application layer.

**Review by thiagohora — COMMENTED** [review]
I'm not sure how bad it's null in ClickHouse so pinged Andres

**Comment by thiagohora** [comment]
Also, can you test the null values won't break the aggregations using this field?

File: apps/opik-backend/src/main/resources/liquibase/db-app-analytics/migrations/000007_change_spans_usage_value_to_nullable.sql (hunk 1 above)
Comment (BorisTkachenko) [reply apps/opik-backend/src/main/resources/liquibase/db-app-analytics/migrations/000007_change_spans_usage_value_to_nullable.sql]: Deleted

File: apps/opik-backend/src/main/resources/liquibase/db-app-analytics/migrations/000007_change_spans_usage_value_to_nullable.sql (hunk 2 above)
Comment (BorisTkachenko) [reply apps/opik-backend/src/main/resources/liquibase/db-app-analytics/migrations/000007_change_spans_usage_value_to_nullable.sql]: Removed

**Comment by BorisTkachenko** [comment]
> Also, can you test the null values won't break the aggregations using this field?

@thiagohora Not relevant, fix updated based on https://comet-ml.slack.com/archives/C0768KH9MDJ/p1731427219349329

**Comment by thiagohora** [comment]
Can you just add a happy path test?

File: apps/opik-backend/src/main/resources/liquibase/db-app-analytics/migrations/000007_change_spans_usage_value_to_nullable.sql (hunk 2 above)
Comment (andrescrz) [reply apps/opik-backend/src/main/resources/liquibase/db-app-analytics/migrations/000007_change_spans_usage_value_to_nullable.sql]: We discussed to not to follow this approach. This is an implementation detail which affects performance. We can perfectly fulfil the user requirement with other alternatives without changing the low level schema.

**Comment by BorisTkachenko** [comment]
> Can you just add a happy path test?
@thiagohora 
Added

File: apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/SpansResourceTest.java [hunk 3]
```diff
@@ -3412,6 +3419,41 @@ void batch__whenSendingMultipleSpansWithNoId__thenReturnNoContent() {
...
+                    .path("batch")
+                    .request()
+                    .header(HttpHeaders.AUTHORIZATION, API_KEY)
+                    .header(WORKSPACE_HEADER, TEST_WORKSPACE)
+                    .post(Entity.json(JsonUtils.getJsonNodeFromString(body)))) {
+
+                assertThat(actualResponse.getStatusInfo().getStatusCode()).isEqualTo(204);
+                assertThat(actualResponse.hasEntity()).isFalse();
+            }
+        }
```
Comment (thiagohora) [inline apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/SpansResourceTest.java]: There is no need for this new ObjectMapper. Also, we need to assert the response to see if the retrieve is as it's supposed to be.

```

```java
void batch__whenCreateSpansUsageWithNullValue__thenReturnNoContent() {

            String projectName = UUID.randomUUID().toString();

            Map<String, Integer> usage = new LinkedHashMap<>() {{
                put("firstKey", 10);
            }};

            var expectedSpans = PodamFactoryUtils.manufacturePojoList(podamFactory, Span.class).stream()
                    .map(span -> span.toBuilder()
                            .usage(usage)
                            .projectName(projectName)
                            .parentSpanId(null)
                            .feedbackScores(null)
                            .build())
                    .toList();

            var spanBatch = new SpanBatch(expectedSpans);

            JsonNode body = JsonUtils.readTree(spanBatch);

            body.get("spans").forEach(span -> {
                var usageNode = span.get("usage");

                if (usageNode instanceof ObjectNode) {
                    ((ObjectNode) usageNode).set("secondKey", NullNode.getInstance());
                }
            });

            try (var actualResponse = client.target(URL_TEMPLATE.formatted(baseURI))
                    .path("batch")
                    .request()
                    .header(HttpHeaders.AUTHORIZATION, API_KEY)
                    .header(WORKSPACE_HEADER, TEST_WORKSPACE)
                    .post(Entity.json(body))) {

                assertThat(actualResponse.getStatusInfo().getStatusCode()).isEqualTo(204);
                assertThat(actualResponse.hasEntity()).isFalse();
            }

            getAndAssertPage(TEST_WORKSPACE, projectName, List.of(), List.of(), expectedSpans.reversed(), List.of(),
                    API_KEY);
        }

Comment (BorisTkachenko) [reply apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/SpansResourceTest.java]: Done

**Review by thiagohora — APPROVED** [review]
